mdsuite/graph_modules/molecular_graph.py

```diff
@@ -247,14 +256,224 @@ def check_a_in_b(a, b):
         for item in del_list:
             molecules.pop(item)
 
-        if n_molecules is None:
-            return molecules
+        return molecules
+
+    def _perform_isomorphism_tests(self):
+        """
+        Run isomorphism checks to determine whether or not the graphs computed are
+        correct.
+
+        Currently runs the following tests:
+
+        1. Checks that the number of decomposed graphs is equal to the number of
+           expected molecules.
+        2. Checks that the number of particles of each constituent species for each
+           molecule matches that given by the SMILES string or the user provided
+           reference data.
+
+        Returns
+        -------
+
+        """
+        # amount of molecules test
+        self._amount_isomorphism_test()
+        # groups equality test
+        self._molecule_group_equality_isomorphism_test()
+
+    def _amount_isomorphism_test(self):
+        """
+        Test that the amount of computed molecules is equal to the expected amount.
+
+        Returns
+        -------
+        Returns nothing, raises a value error if condition is not met.
+        """
+        log.info("Performing molecule number isomorphism test.")
+        # number of molecules test
+        if self.n_molecules is None:
+            log.info("No molecule amount to check against, skipping test.")
         else:
-            if len(molecules) != n_molecules:
+            if len(self.molecular_groups) != self.n_molecules:
                 raise ValueError(
-                    "Expected number of molecules does not "
-                    "match the amount computed, please adjust"
-                    "parameters."
+                    f"Expected number of molecules ({self.n_molecules}) does not "
+                    f"match the amount computed ({len(self.molecular_groups)}), "
+                    "please adjust cutoff parameters."
                 )
             else:
-                return molecules
+                log.info("Amount of molecules test passed.")
+
+    def _molecule_group_equality_isomorphism_test(self):
+        """
+        Test that the molecule groups computed match that of the reference.
+
+        Returns
+        -------
+        Nothing, will raise an exception if the test fails.
+        """
+        log.info("Performing group equality isomorphism test.")
+        for mol_number, mol_data in self.molecular_groups.items():
+            for species, indices in mol_data.items():
+                try:
+                    assert len(indices) == self.species[species]
```

Comment:
assertion should not be used in public API, see https://google.github.io/styleguide/pyguide.html#244-decision

Reply:
Sure I can change that to an exception raise.

Reply:
resolvd abdb92b